docs/ruleset_functions/get_heating_fuels_p.md

```diff
@@ -0,0 +1,75 @@
+        - Reset energy source variable: `energy_source_type_p = ""`
+        - Get the energy_source_type: `energy_source_type_p = heating_systems_p.energy_source_type`
+        - Check if the energy source is propane, if so then set boolean variable to true: `if energy_source_type_p in ["PROPANE"]:fuels_propane_p = TRUE`
+        - Check else if the energy source is natural gas, if so then set boolean variable to true: `elif energy_source_type_p in ["NATURAL_GAS"]:fuels_natural_gas_p = TRUE`
+        - Check else if the energy source is electricity, if so then set boolean variable to true: `elif energy_source_type_p in ["ELECTRICITY"]:fuels_electricity_p = TRUE`
+        - Check else if the energy source is fuel oil, if so then set boolean variable to true: `elif energy_source_type_p in ["FUEL_OIL"]:fuels_oil_p = TRUE`
+        - Else, set other boolean variable to true: `Else:fuels_other_p = TRUE`  
+        Loop through the preheat systems associated with the hvac system.
+        - For each preheat system in hvac_p: `preheat_systems_p in hvac_p.preheat_systems:`
+        - Reset energy source variable: `energy_source_type_p = ""`
+        - Get the energy_source_type: `energy_source_type_p = preheat_systems_p.energy_source_type`
+        - Check if the energy source is propane, if so then set boolean variable to true: `if energy_source_type_p in ["PROPANE"]:fuels_propane_p = TRUE`
+        - Check else if the energy source is natural gas, if so then set boolean variable to true: `elif energy_source_type_p in ["NATURAL_GAS"]:fuels_natural_gas_p = TRUE`
+        - Check else if the energy source is electricity, if so then set boolean variable to true: `elif energy_source_type_p in ["ELECTRICITY"]:fuels_electricity_p = TRUE`
+        - Check else if the energy source is fuel oil, if so then set boolean variable to true: `elif energy_source_type_p in ["FUEL_OIL"]:fuels_oil_p = TRUE`
+        - Else, set other boolean variable to true: `Else:fuels_other_p = TRUE`  
+    Loop through all terminal units and check the heat source associated with each. Boolean variables are set to TRUE accordingly.
+    - For each zone_p in each building_segment_p: `for zone_p in building_segment_p.zones:`
+        - For each terminal_p in zone_p: `for terminal_p in zone_p.terminals:`
+            - Reset energy source variable: `energy_source_type_p = ""`
+            - Get terminal heat source: `energy_source_type_p = terminal_p.heat_source`
+            - Check if the energy source is electric, if so then set boolean variable to true: `if energy_source_type_p in ["ELECTRIC"]:fuels_electricity_p = TRUE`
+            - Check else if the energy source is other, if so then set boolean variable to true (else, do nothing): `elif energy_source_type_p in ["OTHER"]:fuels_other_p = TRUE`  
+This section creates a list of the applicable fuel types that provide heating in the proposed design model based on the outcome of the boolean variables above.
```

Comment:
Per above, we need to be clear whether the list includes only space heating fuels or all fuels.